### PR TITLE
Richer seed events

### PR DESCRIPTION
**Pull request by cloudhead**
Add more information to the seed events.

### CODE REVIEW

File: seed/src/lib.rs [hunk 1]
```diff
@@ -127,11 +130,54 @@ pub enum Event {
...
+        provider: PeerId,
+        name: String,
+        description: Option<String>,
+        maintainers: HashSet<RadUrn>,
+    },
+}
+
+impl Event {
+    async fn peer_connected(peer_id: PeerId, api: &PeerApi<Signer>) -> Result<Self, Error> {
+        let user = self::guess_user(peer_id, api).await?;
```
Comment (FintanH) [inline seed/src/lib.rs:152]: Could move the `.as_ref()` up here?

File: seed/src/lib.rs [hunk 2]
```diff
@@ -317,3 +365,28 @@ impl Node {
...
+    api: &PeerApi<Signer>,
+) -> Result<Option<User<entity::Draft>>, Error> {
+    api.with_storage(move |s| {
+        let metas = s.all_metadata()?;
+
+        for meta in metas {
+            let meta = meta?;
+            let repo = s.open_repo(meta.urn())?;
+
+            for remote in repo.tracked()? {
```
Comment (FintanH) [inline seed/src/lib.rs:381]: You could just do `s.tracked(meta.urn())?`

File: seed/src/lib.rs [hunk 3]
```diff
@@ -317,3 +365,28 @@ impl Node {
         Ok(())
     }
 }
+
+/// Guess a user given a peer id.
+async fn guess_user(
```
Comment (FintanH) [inline seed/src/lib.rs:370]: In the words of @xla, "I don't hate this".

File: seed/src/lib.rs (hunk 1 above)
Comment (cloudhead) [reply seed/src/lib.rs:152]: Sort of, will make a small change.

File: seed/src/lib.rs (hunk 2 above)
Comment (cloudhead) [reply seed/src/lib.rs:381]: Hmm the thing is I use `repo` right below..

Comment (cloudhead) [reply seed/src/lib.rs:381]:
```rust
                    let user = repo.get_rad_self_of(remote)?;
```

File: seed/src/lib.rs (hunk 3 above)
Comment (cloudhead) [reply seed/src/lib.rs:370]: hahaha

File: seed/src/lib.rs (hunk 2 above)
Comment (FintanH) [reply seed/src/lib.rs:382]: Oh aye, I mean you could do `s.get_rad_self_of(meta.urn(), remote)` but I'm splitting hairs at that stage. Disregard :)

**Review by FintanH — APPROVED** [review]
:metal:

**Review by xla — APPROVED** [review]
🛄 📮 🚐 🌙

**Pull request merged**